web/src/stats/stats.ts

```diff
@@ -875,21 +977,33 @@ function populate_number_of_users(data) {
     });
 
     // Initial drawing of plot
-    draw_or_update_plot(all_time_trace, true);
+    // https://github.com/zulip/zulip/commit/fa9d79e203f83ab51b28e79bbb7a11589a2fd591
+    draw_or_update_plot(all_time_trace);
     $("#all_time_actives_button").addClass("selected");
     get_user_summary_statistics(data.everyone);
 }
 
-function populate_messages_read_over_time(data) {
+function populate_messages_read_over_time(data: ReadData): void {
     if (data.end_times.length === 0) {
         // TODO: do something nicer here
         return;
     }
 
     // Helper functions
-    function make_traces(dates, values, type, date_formatter) {
+    function make_traces(
+        dates: Date[],
+        values: DataByEveryoneMe<number[]>,
+        type: Plotly.PlotType,
+        date_formatter: DateFormatter,
+    ): DataByEveryoneMe<Partial<Plotly.PlotData>> {
         const text = dates.map((date) => date_formatter(date));
-        const common = {x: dates, type, hoverinfo: "none", text, textposition: "none"};
+        const common: {
+            x: Plotly.Datum[];
+            type: Plotly.PlotType;
+            hoverinfo: "none";
+            text: string[];
+            textposition: "none";
+        } = {x: dates, type, hoverinfo: "none", text, textposition: "none"};
```

Comment:
We can remove this inline type annotation and rely on type inference for this one.

Reply:
Ideally yes except one variable, `x` here. `x` is `Plotly.Datum[]`, a union type of `Date[]`, because by definition `type Datum = string | number | Date | null;`. Either we need to cast type using `as` keyword, or we keep inline type annotation to avoid type inference.

Reply:
Actually, we can still use type annotation but in a simpler way like [previous lines](https://github.com/zulip/zulip/blob/040d686961aae32a80bbd3fe6228a3d35026efba/web/src/stats/stats.ts#L270-L276) have done:
```
common: Partial<Plotly.PlotData> = {
            x: dates,
            type,
            hoverinfo: "none",
            text,
            textposition: "none",
        };
```